shared/modules/network.utils.ts

```diff
@@ -91,3 +103,50 @@ export function convertNetworkId(value: unknown): string | null {
   }
   return null;
 }
+
+/**
+ * Convert an eip155 CAIP chain ID to a hex chain ID.
+ *
+ * @param id - The CAIP chain ID to convert.
+ * @returns The hex chain ID.
+ */
+export function convertCaipToHexChainId(id: CaipChainId): Hex {
+  const { namespace, reference } = parseCaipChainId(id);
+  if (namespace === KnownCaipNamespace.Eip155) {
+    return `0x${parseInt(reference, 10).toString(16)}`;
```

Comment:
Very nit: We do have a `add0x` in `@metamask/utils` for this: https://github.com/MetaMask/utils/blob/main/src/hex.ts#L121-L138

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/30433/commits/dc0777d13b55902d7b9c939722dc9e4113abbcb7